fuel-core/src/database/sealed_block.rs

```diff
@@ -0,0 +1,100 @@
+use crate::database::{
+    storage::{
+        FuelBlocks,
+        SealedBlockConsensus,
+    },
+    Column,
+    Database,
+};
+use fuel_core_interfaces::{
+    common::prelude::{
+        Bytes32,
+        StorageAsRef,
+        StorageInspect,
+        StorageMutate,
+    },
+    db::KvStoreError,
+    model::{
+        FuelBlockConsensus,
+        SealedFuelBlock,
+        SealedFuelBlockHeader,
+    },
+};
+use std::borrow::Cow;
+
+impl StorageInspect<SealedBlockConsensus> for Database {
+    type Error = KvStoreError;
+
+    fn get(
+        &self,
+        key: &Bytes32,
+    ) -> Result<Option<Cow<FuelBlockConsensus>>, KvStoreError> {
+        Database::get(self, key.as_ref(), Column::FuelBlockConsensus).map_err(Into::into)
+    }
+
+    fn contains_key(&self, key: &Bytes32) -> Result<bool, KvStoreError> {
+        Database::exists(self, key.as_ref(), Column::FuelBlockConsensus)
+            .map_err(Into::into)
+    }
+}
+
+impl StorageMutate<SealedBlockConsensus> for Database {
+    fn insert(
+        &mut self,
+        key: &Bytes32,
+        value: &FuelBlockConsensus,
+    ) -> Result<Option<FuelBlockConsensus>, KvStoreError> {
+        Database::insert(self, key.as_ref(), Column::FuelBlockConsensus, value)
+            .map_err(Into::into)
+    }
+
+    fn remove(
+        &mut self,
+        key: &Bytes32,
+    ) -> Result<Option<FuelBlockConsensus>, KvStoreError> {
+        Database::remove(self, key.as_ref(), Column::FuelBlockConsensus)
+            .map_err(Into::into)
+    }
+}
+
+impl Database {
+    pub fn get_sealed_block(
```

Comment:
Should we check for mismatches here? Like the header having a different consensus type then the `FuelBlockConsensus`?

Reply:
we could, although there's not really much the application can do if it gets into that state 🤔

Reply:
I think we should defer this checking until we have more than one kind of consensus variant, as rust won't let me even write the error case right now.